kaldb/src/main/java/com/slack/kaldb/metadata/service/ServicePartitionMetadata.java

```diff
@@ -0,0 +1,88 @@
+package com.slack.kaldb.metadata.service;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.collect.ImmutableList;
+import com.slack.kaldb.proto.metadata.Metadata;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Metadata for a specific partition configuration at a point in time. For partitions that are
+ * currently active we would expect to have an endTime of max long.
+ */
+public class ServicePartitionMetadata {
```

Comment:
Do we need a separate class for this message? Can we include this in service metadata itself? 

Also, since this is an internal struct, may we can just call this a partition map and not a metadata?

If we need it to be a separate class can it be an internal static class in ServiceMetadata?

Reply:
[Per previous feedback](https://github.com/slackhq/kaldb/pull/144#discussion_r747761893) it was requested that we avoid nesting where possible, due to limited namespace support. If we intend to change guidance for this we should revisit the other protobuf definition files that also largely conform to this pattern (ie `kaldb_configs`).

I'm not a fan of calling this a partition map, as it's not a map. This makes it unnecessarily confusing when trying to consume the object as you end up with code like `partitionMap.stream()` or `partitionMap.sort()`